test/EIP7702Proxy/initialize.t.sol

```diff
@@ -149,79 +247,53 @@ contract InitializeTest is EIP7702ProxyBase {
-        vm.assume(newOwner != mockPreviousImpl);
-        vm.assume(newOwner != _eoa);
-        assumeNotPrecompile(mockPreviousImpl);
-        assumeNotPrecompile(newOwner);
-        vm.assume(uninitProxyPk != 0);
-        vm.assume(uninitProxyPk != _EOA_PRIVATE_KEY);
-
-        // Derive address for uninitProxy from private key
-        address payable uninitProxy = payable(vm.addr(uninitProxyPk));
-
-        // Deploy proxy template and etch its code at the target address
-        EIP7702Proxy proxyTemplate = new EIP7702Proxy(
-            address(_implementation),
-            _initSelector
-        );
-        bytes memory proxyCode = address(proxyTemplate).code;
-        vm.etch(uninitProxy, proxyCode);
-
-        // Set the implementation slot to some other address, simulating a previous implementation
-        vm.store(
-            uninitProxy,
-            ERC1967Utils.IMPLEMENTATION_SLOT,
-            bytes32(uint256(uint160(mockPreviousImpl)))
-        );
+    function test_reverts_whenSignatureUsesWrongNonce() public {
+        // Get current nonce
+        uint256 currentNonce = _nonceTracker.getNextNonce(_eoa);
 
-        // Verify implementation slot is set to the previous implementation
-        assertEq(
-            _getERC1967Implementation(uninitProxy),
-            mockPreviousImpl,
-            "Implementation slot should be set to previous implementation"
-        );
-
-        // Initialize the proxy
+        // Create signature with wrong (future) nonce
         bytes memory initArgs = _createInitArgs(_newOwner);
         bytes32 initHash = keccak256(
-            abi.encode(address(proxyTemplate), initArgs)
+            abi.encode(_proxy, initArgs, currentNonce + 1)
         );
-        (uint8 v, bytes32 r, bytes32 s) = vm.sign(uninitProxyPk, initHash);
+        (uint8 v, bytes32 r, bytes32 s) = vm.sign(_EOA_PRIVATE_KEY, initHash);
         bytes memory signature = abi.encodePacked(r, s, v);
 
-        EIP7702Proxy(uninitProxy).initialize(initArgs, signature);
+        vm.expectRevert(EIP7702Proxy.InvalidSignature.selector);
+        EIP7702Proxy(_eoa).initialize(initArgs, signature);
+    }
 
-        // Verify implementation slot was changed to the correct implementation
-        assertEq(
-            _getERC1967Implementation(uninitProxy),
-            address(_implementation),
-            "Implementation slot should be set to correct implementation"
+    function test_reverts_whenSignatureReplayedWithSameNonce() public {
+        // First initialization
+        bytes memory initArgs = _createInitArgs(_newOwner);
+        bytes memory signature = _signInitData(_EOA_PRIVATE_KEY, initArgs);
+        EIP7702Proxy(_eoa).initialize(initArgs, signature);
+
+        // Try to replay the same signature
+        vm.expectRevert(EIP7702Proxy.InvalidSignature.selector);
+        EIP7702Proxy(_eoa).initialize(initArgs, signature);
+    }
+
+    function test_constructor_reverts_whenImplementationZero() public {
```

Comment:
would recommend having separate `constructor.t.sol` tests and also happy path cases for sanity check and 100% coverage